geoportailv3/views/luxprintproxy.py

```diff
@@ -83,8 +85,9 @@ def lux_report_create(self):
 
     @view_config(route_name="lux_printproxy_report_cancel")
     def lux_cancel(self):
-        ref = self.request.matchdict.get("ref")
-        del self.ref_spec[ref]
+        DBSession.query(LuxPrintJob).filter(
+            LuxPrintJob.ref == self.request.matchdict.get("ref")
+        ).delete()
```

Comment:
I don't know if it's more efficient or not. But I find it a bit more elegant. ;-)